### PR TITLE
adding Gemis 2021

### PR DESCRIPTION
**Pull request by AnastasiaLobanova**
**Is this a new addition or an update to an existing factor?**

- [x] New
- [ ] Update

**What category or categories does it affect?**

multiple

**What?**

adding Gemis 2021

**Why?**

provide better coverage for DE sources

### CODE REVIEW

**Review by maryarbab — CHANGES_REQUESTED** [review]
Looks great! There are bunch of commas in the description, which might be better to remove to avoid formatting issues in the G sheets.

**Review by patstirling — CHANGES_REQUESTED** [review]
great dataset - however it is an extremely varied one across not that many EFs so there's a lot of comments!

File: OpenEmissionFactorsDB.csv [hunk 1]
```diff
@@ -2843,7 +2865,20 @@ f307212a-9bc9-4a2e-8cfb-924a48248efb,Energy,Electricity,electricity-energy_sourc
...
+,Energy,Electricity,electricity-energy_source_electricity_grid_local,Electricity grid local,KWh,0.56,not-supplied,0.531,0.0006,0.00004,not-supplied,not-supplied,,,upstream-use_phase-transport-transmission_and_distribution,GEMISv5.0,2021,2021,2015,DE,,Climatiq,2022/06/20,Emission intensity of local power grid (low voltage) households/small consumption. incl. grid losses. Total life cycle incl. transport + material input without disposal,https://iinas.org/downloads/gemis-downloads/
+,Energy,Electricity,electricity-energy_source_electricity_power_plant_mix,Electricity power plant mix,KWh,0.543,not-supplied,0.515,0.00058,0.00004,not-supplied,not-supplied,,,upstream-use_phase-transport,GEMISv5.0,2021,2021,2015,DE,,Climatiq,2022/06/20,Emission intensity of power plant mix. Without power transport/distribution and substation losses. Total life cycle incl. transport + material input without disposal,https://iinas.org/downloads/gemis-downloads/
```
[the remark this answers is not in the document]
Comment (AnastasiaLobanova) [reply OpenEmissionFactorsDB.csv:2879]: the "grid mix" are from statistical data and one from 2015 are calculated, as I understood

File: OpenEmissionFactorsDB.csv [hunk 2]
```diff
@@ -2843,7 +2865,20 @@ f307212a-9bc9-4a2e-8cfb-924a48248efb,Energy,Electricity,electricity-energy_sourc
...
+,Energy,Electricity,electricity-energy_source_electricity_grid_local,Electricity grid local,KWh,0.6,not-supplied,0.569,0.00075,0.00003,not-supplied,not-supplied,,,upstream-electricity_generation-transmission_and_distribution,GEMISv5.0,2021,2021,2010,DE,,Climatiq,2022/06/20,Emission intensity of local power grid (low voltage) for households/small consumption incl. grid losses (according to statistical data).. Total life cycle incl. transport + material input without disposal,https://iinas.org/downloads/gemis-downloads/
```
[the remark this answers is not in the document]
Comment (AnastasiaLobanova) [reply OpenEmissionFactorsDB.csv:2868]: This is statistical data from GEMIS take a look at years calculated from - they have separately grid mix and grid local, I would keep this this way to align with the source

File: OpenEmissionFactorsDB.csv [hunk 3]
```diff
@@ -2957,6 +2992,16 @@ edaa8b40-487b-41b0-9f88-906464411f33,Energy,Electricity,electricity-energy_sourc
...
 63d1dccf-b19b-44f0-8725-cfe7c3c592de,Energy,Electricity,electricity-energy_source_grid_mix,Grid mix,kWh,,,0.461,,,,,10,,electricity_generation,ADEME,2021,,,DE,,Climatiq,2021/08/13,Emission intensity of grid mix as reported by ADEME. The values include direct emissions from power plants. Cross-border power exchanges are not included. Originally provided by the IEA in 2013. Note that values are reported as CO2e by ADEME and as CO2 by the original source.,https://bilans-ges.ademe.fr/en/basecarbone/donnees-consulter/liste-element/categorie/68
+,Energy,Electricity,electricity-energy_source_grid_mix,Grid mix,KWh,0.582,not-supplied,0.552,0.00072,0.00003,not-supplied,not-supplied,,,upstream-electricity_generation,GEMISv5.0,2021,2021,2010,DE,,Climatiq,2022/06/20,Emission intensity of power plant mix (according to statistical data). Excluding electricity transport/distribution and substation losses. Total life cycle incl. transport + material input without disposal,https://iinas.org/downloads/gemis-downloads/
```
[the remark this answers is not in the document]
Comment (AnastasiaLobanova) [reply OpenEmissionFactorsDB.csv:2995]: they vary in the year of calculation - this is statistical data from GEMIS

File: OpenEmissionFactorsDB.csv [hunk 4]
```diff
@@ -5671,13 +5731,50 @@ dcff1f38-af48-4455-8288-f606ef5d517d,Energy,Fuel,fuel_type_wood_pellets-fuel_use
...
+,Energy,Heat and Steam,heat-and-steam-type_district_heating_mix,District heating mix,KWh,0.254,not-supplied,0.237,0.00041,0.00002,not-supplied,not-supplied,,,upstream-use_phase-transport,GEMISv5.0,2021,2021,2015,DE,,Climatiq,2022/06/20,Emission intensity of input-related heat supply by electricity (power plant mix). Final energy related with 100% utilization rate. Scenario output for heating and process heat systems - end-energy related. All heating and process heat suppliers were assumed to have a 100% utilization rate to relate emissions to final energy use. Total life cycle incl. transport + material input without disposal,https://iinas.org/downloads/gemis-downloads/
+,Energy,Heat and Steam,heat-and-steam-type_district_heating_mix,District heating mix,KWh,0.254,not-supplied,0.237,0.00041,0.00002,not-supplied,not-supplied,,,upstream-use_phase-transport,GEMISv5.0,2021,2021,2015,DE,,Climatiq,2022/06/20,Emission intensity of mix of district heating from CHP and heating plants incl. district heating network/line losses/pumping current. Total life cycle incl. transport + material input without disposal,https://iinas.org/downloads/gemis-downloads/
```
[the remark this answers is not in the document]
Comment (AnastasiaLobanova) [reply OpenEmissionFactorsDB.csv]: These two values are identical - I think it is the same ef, just differently described (I think we need to delete the one in 5742, agree?

[the remark this answers is not in the document]
Comment (patstirling) [reply OpenEmissionFactorsDB.csv]: Hm - even though the EFs are the same, concern is that someone using GEMIS previously might expect to see both - one for "district heating" and one for "CHP & district heating"

File: OpenEmissionFactorsDB.csv [hunk 5]
```diff
@@ -7321,6 +7429,9 @@ c5debdfd-62bf-4d11-9285-1c21c624af80,Materials and Manufacturing,Metals,metals-t
...
+,Materials and Manufacturing,Metals,metals-type_aluminium_secondary,Aluminium secondary,kg,1.683,not-supplied,1.48,0.0019,0.00007,not-supplied,not-supplied,,,upstream-manufacturing-transport,GEMISv5.0,2021,2021,2015,DE,,Climatiq,2022/06/20,"Emission intensity of provision of secondary aluminium free works, only production in Germany. This emission factor reflects the manufacturing effort for the provision of selected metals in Germany in 2015 (average data). Where necessary foreign shares (imports) are included. Total life cycle incl. transport + material input, excl. disposal. Reference point product provision, i.e. without use",https://iinas.org/downloads/gemis-downloads/
```
[the remark this answers is not in the document]
Comment (patstirling) [reply OpenEmissionFactorsDB.csv]: Could we add the brackets to the name "Aluminium (secondary)"

File: OpenEmissionFactorsDB.csv [hunk 6]
```diff
@@ -7580,6 +7694,10 @@ e2f3b699-2a62-4e4b-82d9-387604cfc88b,Materials and Manufacturing,Metals,metals_p
...
+,Materials and Manufacturing,Metals,metals-type_steel_electric,Steel Electric,kg,0.511,not-supplied,0.473,0.0011,0.00002,not-supplied,not-supplied,,,upstream-manufacturing-transport,GEMISv5.0,2021,2021,2015,DE,,Climatiq,2022/06/20,"Emission intensity of manufacture of steel in Germany, electric steel. This emission factor reflects the manufacturing effort for the provision of selected metals in Germany in 2015 (average data). Where necessary foreign shares (imports) are included. Total life cycle incl. transport + material input, excl. disposal. Reference point product provision, i.e. without use",https://iinas.org/downloads/gemis-downloads/
```
[the remark this answers is not in the document]
Comment (patstirling) [reply OpenEmissionFactorsDB.csv]: Could we add the brackets to name e.g. "Steel (electric)"

File: OpenEmissionFactorsDB.csv [hunk 7]
```diff
@@ -5671,13 +5731,50 @@ dcff1f38-af48-4455-8288-f606ef5d517d,Energy,Fuel,fuel_type_wood_pellets-fuel_use
...
+,Energy,Heat and Steam,heat-and-steam-type_electric_heat_pump_soil,Electric heat pump - Soil,KWh,0.162,not-supplied,0.154,0.00018,0.00001,not-supplied,not-supplied,,,upstream-use_phase-transport,GEMISv5.0,2021,2021,2015,DE,,Climatiq,2022/06/20,Emission intensity of electric heat pump - heat source ground. Electricity from power plant mix. Total life cycle incl. transport + material input without disposal,https://iinas.org/downloads/gemis-downloads/
+,Energy,Heat and Steam,heat-and-steam-type_electric_heat_pump_water,Electric heat pump - Water,KWh,0.148,not-supplied,0.14,0.00017,0.00001,not-supplied,not-supplied,,,upstream-use_phase-transport,GEMISv5.0,2021,2021,2015,DE,,Climatiq,2022/06/20,Emission intensity of electric heat pump - heat source ground water. Electricity from power plant mix. Total life cycle incl. transport + material input without disposal,https://iinas.org/downloads/gemis-downloads/
```
[the remark this answers is not in the document]
Comment (AnastasiaLobanova) [reply OpenEmissionFactorsDB.csv]: I think description here is explanatory here

File: OpenEmissionFactorsDB.csv [hunk 8]
```diff
@@ -5731,45 +5731,45 @@ dcff1f38-af48-4455-8288-f606ef5d517d,Energy,Fuel,fuel_type_wood_pellets-fuel_use
...
+dcd7aaad-2d12-460b-a846-ea730486cd39,Energy,Heat and Steam,heat-and-steam-type_central_heating_chp_lpg,Central heating (LPG-fired CHP),kWh,0.319,not-supplied,0.316,0.0001,0,not-supplied,not-supplied,,,upstream-use_phase-transport,GEMIS,2021,2021,2015,DE,,Climatiq,2022/06/20,Emission intensity of central heating by liquid petroleum gas (liquidized petroleum gas). Heat obtained from combined power and heat (CHP). Only heat-related share is reflected. Total life cycle incl. transport + material input without disposal,https://iinas.org/downloads/gemis-downloads/
+f0e0de67-5ba5-4055-9db5-32a2546cf5bf,Energy,Heat and Steam,heat-and-steam-type_local_heating_chp_biogas,Local heating (biogas-fired CHP),kWh,0.095,not-supplied,0.064,0.00038,0.00008,not-supplied,not-supplied,,,upstream-use_phase-transport,GEMIS,2021,2021,2015,DE,,Climatiq,2022/06/20,Emission intensity of local heat from biogas combined power and heat (CHP) (85% corn and 15% slurry) and gas boiler incl. heat network/line losses and pumping electricity. No GHG emissions from land use change (0luc) included here.. Total life cycle incl. transport + material input without disposal,https://iinas.org/downloads/gemis-downloads/
```
[the remark this answers is not in the document]
Comment (AnastasiaLobanova) [reply OpenEmissionFactorsDB.csv]: Done!

File: OpenEmissionFactorsDB.csv [hunk 9]
```diff
@@ -7580,6 +7694,10 @@ e2f3b699-2a62-4e4b-82d9-387604cfc88b,Materials and Manufacturing,Metals,metals_p
...
+,Materials and Manufacturing,Metals,metals-type_steel_galvanized,Steel galvanized,kg,2.284,not-supplied,2.043,0.0068,0.00014,not-supplied,not-supplied,,,upstream-manufacturing-transport,GEMISv5.0,2021,2021,2015,DE,,Climatiq,2022/06/20,"Emission intensity of production of steel sheet in Germany, mix oxygen and electrical steel. This emission factor reflects the manufacturing effort for the provision of selected metals in Germany in 2015 (average data). Where necessary foreign shares (imports) are included. Total life cycle incl. transport + material input, excl. disposal. Reference point product provision, i.e. without use",https://iinas.org/downloads/gemis-downloads/
```
[the remark this answers is not in the document]
Comment (AnastasiaLobanova) [reply OpenEmissionFactorsDB.csv]: Oh true! I will correct this - wrong cope paste from my side

**Review by patstirling — CHANGES_REQUESTED** [review]
Nearly there I think!

Line 2879 was "Power plant mix" which is now "Grid mix" but distinguished from another "Grid mix" only by not including t&d in the LCA. I think this is meant to be an average of the emission intensity of electricity leaving power plants across DE (rather than being delivered to the consumer) so think we can revert this to "Power plant mix" and id="electricity-energy_source_power_plant_mix" to make sure it doesn't accidentally get used as a grid mix.

**Review by maryarbab — APPROVED** [review]

**Review by maryarbab — APPROVED** [review]

**Review by patstirling — APPROVED** [review]
Approved! Strictly speaking as it's a new DB we should be adding details to the source table at the same time as upload - will make a note to follow this up. We should also update the validator tables to allow the source (and for some reason some of the lca codes are being flagged too - will add this to the card)

File: OpenEmissionFactorsDB.csv [hunk 10]
```diff
@@ -2876,7 +2876,7 @@ d0bb725d-d7b3-47ff-8c1e-048a9af2d5fa,Energy,Electricity,electricity-energy_sourc
...
 6de8995e-31b8-4416-b8a3-fff9accf916f,Energy,Electricity,electricity-energy_source_grid_mix,Grid mix,kWh,0.425,not-supplied,0.405,0.00038,0.00003,not-supplied,not-supplied,,,upstream-electricity_generation-transmission_and_distribution,GEMIS,2019,2019,2019,DE,,Climatiq,2022/06/20,Emission intensity of local power grid (low voltage) for households/small consumption incl. grid losses (according to statistical data). Total life cycle incl. transport + material input without disposal. Retrieved from GEMIS v5.0.,https://iinas.org/downloads/gemis-downloads/
 4b762ea3-2c88-401b-8e58-e8bb1e816b5f,Energy,Electricity,electricity-energy_source_grid_mix,Grid mix,kWh,0.56,not-supplied,0.531,0.0006,0.00004,not-supplied,not-supplied,,,upstream-use_phase-transport-transmission_and_distribution,GEMIS,2021,2021,2015,DE,,Climatiq,2022/06/20,Emission intensity of local power grid (low voltage) households/small consumption. incl. grid losses. Total life cycle incl. transport + material input without disposal. Retrieved from GEMIS v5.0.,https://iinas.org/downloads/gemis-downloads/
```
Comment (patstirling) [inline OpenEmissionFactorsDB.csv:2878]: ah - hang on, we'd better replace all instances of ´transport´with ´transport_and_delivery´, as the former doesn't exist in our current schema of lca_activity. Could you confirm this is a sensible change @maryarbab ?

**Review by patstirling — CHANGES_REQUESTED** [review]
The validator is failing on ah - lca_activity = ´transport´- think this needs to be ´transport_and_delivery´. Could you confirm @maryarbab ?

**Comment by GeeWee** [comment]
Remember to add `GEMIS` to the list in the schema - I can see CI is failing

**Comment by patstirling** [comment]
> 

yes, we may as well do this at the same time now (and update [OEFDB_sources.md](https://github.com/climatiq/Open-Emission-Factors-DB/blob/main/OEFDB_sources.md))

**Comment by GeeWee** [comment]
I would suggest as a best-practice you don't let anything into the `main` branch that doesn't pass CI.
Let's keep the documents separate for now, and just try to remember to update both when needed.

**Review by patstirling — APPROVED** [review]

**Pull request merged**